Add Next.js route protection example with `auth().orgId`

LauraBeatris opened the pull request:
### 🔎 Previews:

-

### What does this solve?

- <!--- Describe your changes in detail. Why does this change need to happen? Include any links to Slack discussions, Linear comments, etc. -->

### What changed?

- <!--- How does this PR solve the problem? -->

### Checklist

- [ ] I have clicked on "Files changed" and performed a thorough self-review
- [ ] I have added the "deploy-preview" label and added the preview link(s) to this PR description
- [ ] All existing checks pass

alexisintech commented:
I changed the explanation because it wasn't accurate. It wasn't showing how to "prevent users from creating resources for other organizations they don't belong to." An example for that would have involved multiple organizations, and then using the user's org ID to only perform actions for the organization that matched that ID. But the example that was provided doesn't do that - so I updated the explanation to better fit the example provided. Let me know if you have any questions!

alexisintech reviewed (approved):
😸💖

LauraBeatris commented:
> I changed the explanation because it wasn't accurate. It wasn't showing how to "prevent users from creating resources for other organizations they don't belong to." An example for that would have involved multiple organizations, and then using the user's org ID to only perform actions for the organization that matched that ID. But the example that was provided doesn't do that - so I updated the explanation to better fit the example provided. Let me know if you have any questions!

@alexisintech that's correct, thanks for adjusting it!
